src/webui/www/public/scripts/client.js

```diff
@@ -348,6 +350,16 @@ window.addEvent('load', function () {
                         updateCategoryList();
                         torrentsTableContextMenu.updateCategoriesSubMenu(category_list);
                     }
+
+                    // re-select the previously selected rows
+                    if (full_update) {
+                        if (torrentsTableSelectedRows.length)
+                            torrentsTable.selectRow(torrentsTableSelectedRows.pop());
```

Comment:
I can't figure out what this is doing, do you mind explain a bit?

Reply:
Essentially it selects the first row, clearing all previous selections. It then selects all remaining rows but without clearing the previous selections.

I'll clean this up a bit for readability.

Reply:
I see it now, thanks.
I think its better to remove `this.selectedRows.empty();` in `selectRow()` to simplify usage:
```javascript
if (full_update)
    torrentsTable.deselectAll();

torrentsTableSelectedRows.each(function(rowId) {
    torrentsTable.selectRow(rowId);
});
```

Reply:
I don't believe that's more straightforward because it will require prepending a call to `deselectAll()` to each existing instance of `selectRow()`. I think `selectRow()` should continue to handle the decision to clear existing selections.

Also, the logic in that snippet wouldn't work because we only reselect rows on a full update. I've just pushed a change which incorporates a `selectRows()` method (but still uses the `clear` variable). Let me know your thoughts.